Code/src/main/java/nl/utwente/group10/haskell/type/ConstT.java

```diff
@@ -69,4 +70,26 @@ public String toString() {
             return this.constructor;
         }
     }
+
+    @Override
+    public int compareTo(final Type type) {
+        if (type instanceof ConstT) {
+            if (this.constructor.compareTo(((ConstT) type).constructor) == 0) {
```

Comment:
For String, you could probably use `equals` instead of `compareTo() == 0`:

> compareTo returns 0 exactly when the equals(Object) method would return true.